### ISSUE TITLE
sms sending is blocked by pending (secured) messages

### ISSUE THREAD

**Issue by bdribault** [signalapp/Signal-Android#10594]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
SMS (unsecured messages) can't be sent if a secured message is waiting to be sent. This happens if the device does not have internet access.

### Steps to reproduce
- Open the app
- Open a conversation with a contact
- Disable WIFI and mobile data. (You should not have access to internet)
- Send a (secured) message. This message won't be delivered since the data network is disabled.
- Send an unsecured message (SMS).

**Actual result:** The SMS is not delivered.
**Expected result:** The SMS should be delivered (the GSM network is available).

### Device info

**Device:** Xiaomi Redmi Note 8 Pro
**Android version:** 9 (V11.0.6.0.PGGMIXM, PPR1.180610.011)
**Signal version:** 5.0.8

### Link to debug log

https://debuglogs.org/52b1062bc92b9657578cc30798eb1c7137083fdd1bf74a212987154cff499830

**Comment by a-d-collins** [comment]
@bdribault, is there any chance that the unsecured sms message was in fact sent without you being notified of the success?

I'm new to this codebase, but the debug log contains the following line that lends me to believe that the message may have been sent:
> 01-13 16:15:10.931 18589 10210 D EventBus: No subscribers registered for event class org.thoughtcrime.securesms.sms.MessageSender$MessageSentEvent

**Comment by bdribault** [comment]
Thanks you @a-d-collins for investigating.
None of the messages are received on the other device while my device is disconnected from internet (I have the two devices within reach).
As soon as I connect back to internet, all the messages are received on the other device, and marked as received on my device.

**Comment by Tursko** [comment]
Just encountered this due to the server outage.

https://debuglogs.org/6fd9068fa4bfb0e79a370cf461d566416c263b33dac3d3adb9553262e5ffe1cb

**Comment by kyleedwardsny** [comment]
+1, this is happening to me too. Very irritating that I can't use SMS while the servers are down and I have outgoing Signal messages.

**Comment by saucoide** [comment]
Tried this with the same results on 2 different phones (both xiaomi though), SMS stays pending until there is internet connection, deleting the pending messages has no effect

This is the same as this one https://github.com/signalapp/Signal-Android/issues/9864, can they be merged?

**Comment by Tursko** [comment]
Duplicate of #9864

**Comment by Tursko** [comment]
This is not relevant anymore. This is a duplicate and the bug was already fixed.
